app/src/main/java/com/arielwang/workoutlogger/features/workoutaddingflow/exercise/ui/screen/ExerciseViewModel.kt

```diff
@@ -66,7 +77,7 @@ class ExerciseViewModel @Inject constructor(
                 navigator.navigate(TrackDestination.route())
             }
             is ExerciseView.Action.GoBackToPreviousPage -> {
-                navigator.navigateUp()
+                navigator.navigate(HomeDestination.route())
```

Comment:
Should be navigateUp here. The difference is that navigate will put another screen into the navigation stack